x-pack/plugins/session_view/public/components/session_view_search_bar/index.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useState, useEffect } from 'react';
+import { EuiSearchBar, EuiPagination } from '@elastic/eui';
+import { EuiSearchBarOnChangeArgs } from '@elastic/eui';
+import { Process } from '../../../common/types/process_tree';
+import { useStyles } from './styles';
+
+interface SessionViewSearchBarDeps {
+  searchQuery: string;
+  setSearchQuery(val: string): void;
+  searchResults: Process[] | null;
+  setSelectedProcess(process: Process): void;
+}
+
+/**
+ * The main wrapper component for the session view.
+ */
+export const SessionViewSearchBar = ({
+  searchQuery,
+  setSearchQuery,
+  setSelectedProcess,
+  searchResults,
+}: SessionViewSearchBarDeps) => {
+  const styles = useStyles();
+
+  const [selectedResult, setSelectedResult] = useState(0);
+
+  const onSearch = ({ query }: EuiSearchBarOnChangeArgs) => {
+    setSelectedResult(0);
+
+    if (query) {
+      setSearchQuery(query.text);
+    } else {
+      setSearchQuery('');
+    }
+  };
+
+  const renderSearchResults = () => {
+    if (searchResults?.length) {
+      return (
+        <EuiPagination
+          data-test-subj="searchPagination"
+          css={styles.pagination}
+          pageCount={searchResults.length}
+          activePage={selectedResult}
+          onPageClick={setSelectedResult}
+          compressed
+        />
+      );
+    }
+  };
```

Comment:
Same here. Could you explain why an extra render function is used here instead of inlining the JSX? It feels counter-intuitive to me but maybe I'm missing sth here.

Reply:
are you saying you'd prefer an inline check like

{searchResults?length && <EuiPagination
 ...
 />}
 
 instead of a function?

Reply:
Yep, something along the lines of:

```
const showPagination = searchResults?.length;

(...)

{showPagination && <EuiPagination
(...)
/>}
```

Reply:
updated